GVRf/Framework/src/org/gearvrf/GVRViewManager.java

```diff
@@ -246,27 +260,154 @@ void beforeDrawEyes() {
+            int width = 0, height = 0;
+            if (mScreenshotCenterCallback != null
+                    || mScreenshotLeftCallback != null
+                    || mScreenshotRightCallback != null) {
+                GVRRenderTexture rendernTexture = mRenderBundle.getRightRenderTexture();
+                width = rendernTexture.getWidth();
+                height = rendernTexture.getHeight();
+                readbackBuffer = ByteBuffer.allocateDirect(width * height * 4);
+                readbackBuffer.order(ByteOrder.nativeOrder());
+            }
+
             if (eye == 1) {
                 mainCameraRig.predict(4.0f / 60.0f);
                 GVRCamera rightCamera = mainCameraRig.getRightCamera();
                 renderCamera(mActivity.appPtr, mMainScene, rightCamera,
                         mRenderBundle.getRightRenderTexture(), mRenderBundle);
+
+                // if mScreenshotRightCallback is not null, capture right eye
+                if (mScreenshotRightCallback != null) {
+                    readRenderResult(mRenderBundle
+                            .getPostEffectRenderTextureA().getPtr(),
+                            readbackBuffer);
+                    returnScreenshotToCaller(mScreenshotRightCallback,
+                            readbackBuffer, width, height);
+
+                    mScreenshotRightCallback = null;
+                }
+
                 mActivity.setCamera(rightCamera);
             } else {
                 mainCameraRig.predict(3.5f / 60.0f);
+
+                // if mScreenshotCenterCallback is not null, capture center eye
+                if (mScreenshotCenterCallback != null) {
+
+                    // temporarily create a center camera
+                    GVRCamera centerCamera = new GVRPerspectiveCamera(this);
+                    centerCamera.setRenderMask(GVRRenderMaskBit.Left
+                            | GVRRenderMaskBit.Right);
+                    GVRSceneObject centerCameraObject = new GVRSceneObject(this);
+                    centerCameraObject.attachCamera(centerCamera);
+                    mainCameraRig.getOwnerObject().addChildObject(
+                            centerCameraObject);
+
+                    renderCamera(mActivity.appPtr, mMainScene, centerCamera,
+                            mRenderBundle.getRightRenderTexture(),
+                            mRenderBundle);
+                    
+                    centerCameraObject.detachCamera();
+                    mainCameraRig.getOwnerObject().removeChildObject(
+                            centerCameraObject);
+
+                    readRenderResult(mRenderBundle
+                            .getPostEffectRenderTextureA().getPtr(),
+                            readbackBuffer);
+                    returnScreenshotToCaller(mScreenshotCenterCallback,
+                            readbackBuffer, width, height);
+
+                    mScreenshotCenterCallback = null;
+                }
```

Comment:
Actually capture center's logic lies in when we draw left eye buffer, so that the result may be a little bit left. I tried to move this codes block where we draw right eye buffer, and the result will be a little bit right. I talked to Guodong and he said it may be related to the difference on the predict value on mainCameraRig.predict(). 

Anyway, I think this is actually acceptable. Just to see if there is any better solution to get camera angle rather than directly setRenderMask and attach to mainCameraRig.